packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx

```diff
@@ -547,7 +547,7 @@ const TinyMCE = (props: NoteBodyEditorProps, ref: any) => {
 		//
 		// tl;dr: editorReady is used here because the css needs to be re-applied after TinyMCE init
 		// eslint-disable-next-line @seiyab/react-hooks/exhaustive-deps -- Old code before rule was applied
-	}, [editorReady, props.themeId, lightTheme, props.whiteBackgroundNoteRendering]);
+	}, [editorReady, props.themeId, lightTheme, props.whiteBackgroundNoteRendering, props.watchedNoteFiles]);
```

Comment:
Because `props.watchedNoteFiles` isn't used in the body of that hook, consider leaving a comment about why `props.watchedNoteFiles` is included in the dependency array.

Other than that, this pull request looks good to me!

Alternatively, it could make more sense to depend on `styles` instead of `props.watchedNoteFiles`.

Reply:
I added Comment , additionally I want to know `watchedNoteFiles` is used for checking if external editor is on or off but if there is anything which keeps track of each notes state If that note is open in external editor?

Reply:
> is anything which keeps track of each notes state If that note is open in external editor?

[ExternalEditWatcher.ts](https://github.com/laurent22/joplin/blob/dev/packages/lib/services/ExternalEditWatcher.ts) is responsible for watching for file updates. The note is updated here:

https://github.com/laurent22/joplin/blob/7638bdf171ead3b8ac268bf79d9f2275af6ecb42/packages/lib/services/ExternalEditWatcher.ts#L144-L145

The `noteChange` event is handled here:

https://github.com/laurent22/joplin/blob/7638bdf171ead3b8ac268bf79d9f2275af6ecb42/packages/app-desktop/gui/NoteEditor/NoteEditor.tsx#L346

What I think then happens is `props.content` is updated, which causes the rich text editor to update its content, near this comparison:

https://github.com/laurent22/joplin/blob/7638bdf171ead3b8ac268bf79d9f2275af6ecb42/packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx#L906

Reply:
> Alternatively, it could make more sense to depend on `styles` instead of `props.watchedNoteFiles`.

I think that would be the better fix - `styles` is in use in this effect but not in the dependencies. This change may be a good opportunity to clean up this effect and remove the legacy `eslint-disable-next-line @seiyab/react-hooks/exhaustive-deps` (which allows this new change to pass the linter even though it shouldn't). It will require extensive testing though @JanhaviAlekar

Reply:
@JanhaviAlekar please let me know if you're interested in completing this pull request. I assume that changing the dependency to `styles` is not too complex.

Reply:
> > @JanhaviAlekar please let me know if you're interested in completing this pull request. I assume that changing the dependency to `styles` is not too complex.
> 
> Apologies for the delay. I was experiencing some laptop issues. I've made the required changes, and now only the testing part is left. I'll test it today.

[packages/app-desktop/gui/NoteEditor/NoteBody/TinyMCE/TinyMCE.tsx](https://github.com/laurent22/joplin/pull/10069/files/6d6084304649dbbf33c955b0a2272b73db0392e5#diff-910138bf94c934ff21769a3dab9a765e3e0841b4a08fb8c295de183d048ff94a)

`[editorReady, props.themeId, lightTheme, props.whiteBackgroundNoteRendering, styles]);`

I'd like to understand why lightTheme is included in the dependency array. When I removed the `//eslint-disable-next-line @seiyab/react-hooks/exhaustive-deps` , the linter raised an error indicating that `lightTheme` is a dependency not in use. I then removed it, and the code still worked fine. However, I'm uncertain about its purpose or necessity. Could you please provide clarification?

Reply:
Ok then I don't know, we're probably going to break something somewhere if we remove it so let's leave and go back to the original idea. Just include `props.watchedNoteFiles` please and add a comment explaining why it's needed

Reply:
> Ok then I don't know, we're probably going to break something somewhere if we remove it so let's leave and go back to the original idea. Just include `props.watchedNoteFiles` please and add a comment explaining why it's needed

Done! Included props.watchedNoteFiles and added a comment explaining why it's needed. Thank you for your help!